### PR TITLE
Add "Polished" extension

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

Hello!

This is adding https://polished.js.org/ to Raycast. It's also the first time for me writing a Raycast extension so I'm open to any suggestions.

I haven't added all available methods for now and focused on four commands:

- Convert HEX. You can enter a HEX value and get back rgb(a) and HSL
- Convert RGB. You can enter RGB and get back HEX and HSL
- Contrast Ratio. Tells you the ratio between text color and background color
- Color Utilities. You enter a color and can modify a bit + get back more information

I created the App Icon on my own in Figma since we can't use the apple emoji that Polished itself is using.

## Type of change

- New extension

## Screencast

https://user-images.githubusercontent.com/16143594/146679687-efd8a0ac-ca63-471c-9349-2832d45dc1b3.mp4

## Checklist

- [X] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [X] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

File: extensions/polished/src/convert-hex.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  hex: string
+  alpha: string
+}
+
+export default function ConvertHex() {
+  const { push } = useNavigation()
+
+  function handleSubmit(values: Values): void {
+    const { hex, alpha = "1" } = values
+    const a = parseFloat(alpha)
```
Comment (fe9lix) [inline extensions/polished/src/convert-hex.tsx:15]: Looks like the valid range for the alpha component should be validated here. When testing the extension locally here, I run into an assertion when converting colors (I guess at the point where the color would be shown in the list...).

Comment (fe9lix) [reply extensions/polished/src/convert-hex.tsx:15]: Please also check the other commands where an invalid alpha value could be entered.

**Review by fe9lix — CHANGES_REQUESTED** [review]
Hey @LekoArts – thanks for your contribution! I think some of the form element titles could be shortened or moved to the placeholders (will look nicer). And there are some issues with invalid values that I can still enter; they lead to assertion failures when showing the list. Comments inline...

File: extensions/polished/src/color-utilities.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    push(<Results input={inputText} original={parsedColor} lighten={lightenResult} darken={darkenResult} transparentize={transparentizeResult} complement={complement(color)} grayscale={grayscale(color)} invert={invert(color)} readableColor={readableColor(color)} />)    
+  }
+
+  return (
+    <Form actions={
+      <ActionPanel>
+        <SubmitFormAction title="Convert" onSubmit={handleSubmit} />
+      </ActionPanel>
+    }>
+      <Form.TextField id="color" placeholder="#000 or rgb(255, 255, 255) or black" title="Color (HEX, RGB, RGBA, or CSS Color)" />
```
[the remark this answers is not in the document]
Comment (LekoArts) [reply extensions/polished/src/color-utilities.tsx:64]: Thanks for the suggestion! While I'm not 100% happy with that solution it's probably the best thing for the moment. Ideally there would be a "description" field for a form field but that doesn't exist yet :)

**Comment by LekoArts** [comment]
Thanks! Made changes according to your review comments :)

**Review by fe9lix — APPROVED** [review]

**Pull request merged**